src/App.jsx

```diff
@@ -6,25 +6,31 @@ import TodoList from './TodoList'
 import AddTodoForm from './AddTodoForm'
 
 
-
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = useState(
-    JSON.parse(localStorage.getItem('savedTodoList')) || []
-  )
-
+function App() {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  useEffect(() => {
+    const getAsyncTodoList = new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({
+          data: {
+            todoList: []  
```

Comment:
The resolved data contained in `todoList` should come from localStorage. So similar to you did it in the `useSemiPersistentState` function before.